proto/pbsubscribe/subscribe.proto

```diff
@@ -51,9 +51,8 @@ message SubscribeRequest {
 
     // Key is a topic-specific identifier that restricts the scope of the
     // subscription to only events pertaining to that identifier. For example,
-    // to receive events for a single service, the service's name is
-    // specified as the key. An empty key indicates that all events in the topic
-    // are of interest.
+    // to receive events for a single service, the service's name is specified
+    // as the key.
```

Comment:
Nice update. Just a note to say that this is a change to the original design where we hoped we'd be able to do things like efficiently watch the entire catalog for changes by subscribing to the whole topic with no key.

In practice now they are split by key this is no longer possible but it's not a code path we actually expose anywhere today.

In the future if we require that, the likely path would be to allow `TopicKey` to hold explicit wildcard sentinel values in some or all parts of the key such that you could subscribe to the key `partition1/*/*` and get all update for a partition for example (which in OSS means the whole state store). That would require changing the matching logic in publish a little but otherwise seems like a reasonable approach as and when it's needed.

Reply:
Does this mean `stream.SubscribeRequest` needs to be updated to enforce non-blank keys as well?

Reply:
I think f29769eb77473e056475e83878da52ddd2c2c0dc may do this.